Rename Assay Design Tests

labkey-danield opened the pull request:
#### Rationale
Test renaming assay design in LKS.

#### Related Pull Requests
* https://github.com/LabKey/biologics/pull/2500
* https://github.com/LabKey/sampleManagement/pull/2219

#### Changes
* Made the filterCard a public method in finder page.
* Added a rename gpat assay test.
* Added a rename shared gpat assay test.
* Added test to validate assay creation after folder import.

labkey-danield committed:
Merge branch 'develop' into fb_renameAssayDesignTests

labkey-danield committed:
Test assay design rename in LKS.

labkey-danield committed:
Merge branch 'develop' into fb_renameAssayDesignTests

labkey-danield committed:
Test assay design rename in LKSM.

labkey-danield committed:
Merge branch 'develop' into fb_renameAssayDesignTests

labkey-danield committed:
Merge branch 'develop' into fb_renameAssayDesignTests

labkey-danield committed:
Don't hide the FilterCards class.
Added test for SampleFinder and error conditions.

labkey-danield committed:
Merge branch 'develop' into fb_renameAssayDesignTests

labkey-danield committed:
Merge branch 'develop' into fb_renameAssayDesignTests

labkey-danield committed:
Merge branch 'release23.11-SNAPSHOT' into 23.11_fb_renameAssayDesignTests

labkey-danield committed:
Merge branch 'release23.11-SNAPSHOT' into 23.11_fb_renameAssayDesignTests

labkey-danield committed:
Merge branch 'release23.11-SNAPSHOT' into 23.11_fb_renameAssayDesignTests

labkey-danield committed:
Add test to validate assay creation after archive import.

labkey-danield committed:
Don't reuse file names for assay run data.

XingY commented on src/org/labkey/test/tests/assay/FolderImportAssayRenameTest.java:
```diff
@@ -0,0 +1,152 @@
+    }
+
+    /**
+     * <p>
```
I appreciate the effort to add this test! I was thinking this test scenario can piggyback on one of the existing AssayExportImportTest: modify one of the test to do a simple assay creation after import. Though assay design rename is a new feature, I worry after a while this test (especially with the file name mentioning rename, yet the test doesn't do renaming) will feel odd. It's good to have some coverage, but I think having a dedicated test class for it is probably not necessary and a bit confusing.

labkey-danield committed:
Merge branch 'release23.11-SNAPSHOT' into 23.11_fb_renameAssayDesignTests

labkey-danield committed:
Implementing code review feedback.
Removing FolderImportAssayRenameTest.java and inserting the test into AssayExportImportTest.testExportXarToPipeline.
Adding .withScreenshot to checker call.
Added check for sample finder for "No results" case.

labkey-danield committed:
Add test for custom grid view on RUn and Results page.
Added a getter for grid view menu items.

labkey-danield committed:
Going back to having the export-import test in their own test class. Tying to use other import tests wqs becoming a bigger pain than it is worth.
